### DISCUSSION TITLE
add end() method to stop all communication with the Transport channel

### DISCUSSION THREAD

**Discussion by lathoub** [FortySevenEffects/arduino_midi_library#216] [Ideas]
The library provides a `begin()` method to open the communication with the Transport layer, but no opposite method is provided (`end()`) to close the communication with the Transport layer.

Use-case: an interruption in the Transport protocol, causing a breakdown of the Transport layer can't be fixed by restarting the MIDI object - due to the missing `end()` method.

A restart of the Transport layer could be forced by calling `MIDI.end(); MIDI.begin();`

Impact: all underlying Transport mechanisms have to implement the end() method, so the impact on backward compatibility (on the Transport layers) and dependancy on a specific version of this library is significant and needs to be planned carefully.

**Comment by lathoub** [comment]
Added PR #218 for the end() method for the Serial Transport layer

**Reply by franky47** [reply]
Merged !

**Reply by lathoub** [reply]
Thanks @franky47 

All Transport layers now have the `end()` method, that will need to be called by the MIDI protocol in `MIDI.h`:
```cpp
    void end()
    {
        mTransport.end();
    };
```

The code itself is simple to add, but creates a backward compatibility problem. The Transport layers now become dependent on a specific version of this MIDI library. AFAIK the `depends` clause in the `library.properties` does not include version (does it?).

Possible solution to the above version issue is to sequence the release in order: this library needs to release first, once that is done, then the Transport layers can release, depending on the just released MIDI library. (Nothing is on fire, so no need to rush the release)

Once we agree on an approach, I can issue the PR for the MIDI `end()` method.
